core/src/commonTest/kotlin/com/xebia/functional/xef/conversation/ConversationSpec.kt

```diff
@@ -113,4 +118,61 @@ class ConversationSpec :
 
       messagesSizePlusMessageResponse shouldBe memories.size
     }
+
+    "functionCall should be null when the model doesn't support functions" {
```

Comment:
Don't know if this case is possible:
`model doesn't support functions but the prompt contains a function`.

Just a doubt, great job!!